builtin/game/chat.lua

```diff
@@ -983,23 +983,31 @@ core.register_chatcommand("days", {
 })
 
 core.register_chatcommand("shutdown", {
-	params = S("[<delay_in_seconds> | -1] [reconnect] [<message>]"),
-	description = S("Shutdown server (-1 cancels a delayed shutdown)"),
+	params = S("[<delay_in_seconds> | -1] [-r] [<message>]"),
+	description = S("Shutdown server (-1 cancels a delayed shutdown, -r allows players to reconnect)"),
 	privs = {server=true},
 	func = function(name, param)
 		local delay, reconnect, message
 		delay, param = param:match("^%s*(%S+)(.*)")
 		if param then
 			reconnect, param = param:match("^%s*(%S+)(.*)")
 		end
-		message = param and param:match("^%s*(.+)") or ""
+		local bool_reconnect = false
+		if reconnect == "-r" then
+			bool_reconnect = true
+		else
+			message = reconnect
+		end
+		if not message then
+			message = param and param:match("^%s*(.+)") or ""
+		end
```

Comment:
When `delay` or `reconnect` are not provided (or invalid), the text is currently cut off to the first word.
I suggest better pattern matching, using a separate variable in case matching failed to to omitted arguments.